### ISSUE TITLE
MLP texture does not support for DLMesh

### ISSUE THREAD

**Issue by wangjksjtu** [NVlabs/nvdiffrec#20]
Thank you for the amazing work!!

I tried to enable the mlp texture for DLM - changing this line https://github.com/NVlabs/nvdiffrec/blob/main/train.py#L632 to
```
mat = initial_guess_material(geometry, True, FLAGS)
```

However, when I run the the job, it produces the following error:
```
iter=    0, img_loss=0.020718, reg_loss=0.000000, lr=0.00010, time=606.2 ms, rem=10.10 m
Traceback (most recent call last):
  File "/home/wangjk/programs/nvdiffrec/train.py", line 778, in <module>
    geometry, mat = optimize_mesh(glctx, geometry, mat, lgt, dataset_train, dataset_validate, FLAGS, pass_idx=0, pass_name="mesh_pass", 
  File "/home/wangjk/programs/nvdiffrec/train.py", line 542, in optimize_mesh
    total_loss.backward()
  File "/home/wangjk/anaconda3/envs/torch-ngp/lib/python3.9/site-packages/torch/_tensor.py", line 307, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/home/wangjk/anaconda3/envs/torch-ngp/lib/python3.9/site-packages/torch/autograd/__init__.py", line 154, in backward
    Variable._execution_engine.run_backward(
RuntimeError: Trying to backward through the graph a second time (or directly access saved tensors after they have already been freed). Saved intermediate values of the graph are freed when you call .backward() or autograd.grad(). Specify retain_graph=True if you need to backward through the graph a second time or if you need to access saved tensors after calling backward.
```

Do you have any ideas about this? Really appreciate it ;)

**Comment by jmunkberg** [comment]
Thanks for your feedback, @wangjksjtu ! 

I tested your use case quickly, and could reproduce the issue. A somewhat ugly workaround is to replace the line 

`mat = initial_guess_material(geometry, True, FLAGS)`

with 

```
        kd_min, kd_max = torch.tensor(FLAGS.kd_min, dtype=torch.float32, device='cuda'), torch.tensor(FLAGS.kd_max, dtype=torch.float32, device='cuda')
        ks_min, ks_max = torch.tensor(FLAGS.ks_min, dtype=torch.float32, device='cuda'), torch.tensor(FLAGS.ks_max, dtype=torch.float32, device='cuda')
        nrm_min, nrm_max = torch.tensor(FLAGS.nrm_min, dtype=torch.float32, device='cuda'), torch.tensor(FLAGS.nrm_max, dtype=torch.float32, device='cuda')
        mlp_min = torch.cat((kd_min[0:3], ks_min, nrm_min), dim=0)
        mlp_max = torch.cat((kd_max[0:3], ks_max, nrm_max), dim=0)
        mlp_map_opt = mlptexture.MLPTexture3D(mesh.aabb(base_mesh), channels=9, min_max=[mlp_min, mlp_max])
        mat =  material.Material({'kd_ks_normal' : mlp_map_opt})
        mat['bsdf'] = 'pbr'
```

That way, the material object is created directly in the main file, not returned from the `initial_guess_material` call. 
I suspect we do not properly copy the mlptexture object properly, but have not had time to look into it in detail.

**Comment by jmunkberg** [comment]
Fixed properly in CL 564cad1e67ad78761870fbc0ccf3f4f1f5f31f86

Thanks for reporting this!

**Comment by wangjksjtu** [comment]
Thanks a lot for the prompt reply and quick fixes! @jmunkberg  Really appreciate your great help ;)
